app/services/school_remover.rb

```diff
@@ -61,6 +68,35 @@ def remove_meters!
 
   private
 
+  def all_unlocked_users_are_linked_to_other_schools?
```

Comment:
I think the logic is correct here: any users linked ONLY to this school should be locked. Any users links to other schools do not have to be locked in order to proceed with archiving/removal. And tests cover that.

We shouldn't have any SQL in a service class though, so any db code needs to be on a model.

The `User` already has a `has_other_schools?` method, so you could probably simplify this to finding the unlocked users and rejecting those that don't have other schools.